collector.go

```diff
@@ -201,6 +248,18 @@ func (c *spaceliftCollector) Collect(metricChannel chan<- prometheus.Metric) {
 	metricChannel <- prometheus.MustNewConstMetric(c.currentBillingPeriodUsedPublicSeconds, prometheus.GaugeValue, float64(query.Usage.UsedPublicMinutes*60))
 	metricChannel <- prometheus.MustNewConstMetric(c.currentBillingPeriodUsedSeats, prometheus.GaugeValue, float64(query.Usage.UsedSeats))
 
+	for _, state := range query.Metrics.StacksCountByState {
+		metricChannel <- prometheus.MustNewConstMetric(c.currentStacksCountByState, prometheus.GaugeValue, state.Value, state.Labels[0])
+	}
+
+	for _, state := range query.Metrics.ResourcesCountByDrift {
+		metricChannel <- prometheus.MustNewConstMetric(c.currentResourcesCountByDrift, prometheus.GaugeValue, state.Value, state.Labels[0])
+	}
+	
+	metricChannel <- prometheus.MustNewConstMetric(c.currentAvgStackSizeByResourceCount, prometheus.GaugeValue, float64(query.Metrics.AvgStackSizeByResourceCount[0].Value))
```

Comment:
We are currently always returning one single data point entry for each of those metrics. But just for code safety sake, could you add a check on the slice length here before accessing by index `[0]`.

So just in case one day we change something on the backend, this code will not panic 🙇🏻 

And maybe the same for the labels above. It's very unlikely to happen, but it can't really hurt to be a bit more defensive here IMO.

Reply:
fixed this one too